mongo/src/main/java/com/redhat/lightblue/mongo/crud/MongoLocking.java

```diff
@@ -80,12 +80,8 @@ private boolean acquire(String callerId,String resourceId, Long ttl, Date now, D
             append(COUNT,1).
             append(VERSION,1);
 
-        try {
-            LOGGER.debug("insert: {}",update);
-            coll.insert(update,WriteConcern.SAFE);
-        } catch (MongoException.DuplicateKey e) {
-            return false;
-        }
+        LOGGER.debug("insert: {}", update);
+        coll.insert(update, WriteConcern.ACKNOWLEDGED);
```

Comment:
ACK should be ok, then.

On Thu, Feb 25, 2016 at 1:09 PM, Brandon Vulaj <notifications@github.com>
wrote:

> In mongo/src/main/java/com/redhat/lightblue/mongo/crud/MongoLocking.java
> <https://github.com/lightblue-platform/lightblue-mongo/pull/218#discussion_r54156119>
> :
>
> > @@ -80,12 +80,8 @@ private boolean acquire(String callerId,String resourceId, Long ttl, Date now, D
> >              append(COUNT,1).
> >              append(VERSION,1);
> >
> > -        try {
> > -            LOGGER.debug("insert: {}",update);
> > -            coll.insert(update,WriteConcern.SAFE);
> > -        } catch (MongoException.DuplicateKey e) {
> > -            return false;
> > -        }
> > +        LOGGER.debug("insert: {}", update);
> > +        coll.insert(update, WriteConcern.ACKNOWLEDGED);
>
> SAFE is deprecated for ACK. I can change back if needed.
>
> http://api.mongodb.org/java/current/com/mongodb/WriteConcern.html#SAFE
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lightblue-platform/lightblue-mongo/pull/218/files#r54156119>
> .
>